ada/extensions/src/libadalang-helpers.adb

```diff
@@ -0,0 +1,126 @@
+with Ada.Command_Line;
+with Ada.Text_IO; use Ada.Text_IO;
+
+with GNATCOLL.Projects;  use GNATCOLL.Projects;
+with GNATCOLL.Traces;
+with GNATCOLL.VFS;       use GNATCOLL.VFS;
+
+with Libadalang.Project_Provider; use Libadalang.Project_Provider;
+
+package body Libadalang.Helpers is
+
+   function "+" (S : String) return Unbounded_String
+                 renames To_Unbounded_String;
+   function "+" (S : Unbounded_String) return String renames To_String;
+
+   package body App is
+      procedure Run is
+
+         package String_Vectors is new Ada.Containers.Vectors
+           (Positive, Unbounded_String);
+
+         UFP : Unit_Provider_Reference;
+         --  When project file handling is enabled, corresponding unit provider
+
+         Ctx : Analysis_Context;
+
+         Files : String_Vectors.Vector;
+
+         Units : Unit_Vectors.Vector;
+
+      begin
+         --  Setup traces from config file
+         GNATCOLL.Traces.Parse_Config_File;
+
+         if not Args.Parser.Parse then
+            return;
+         end if;
+
+         Ctx := Create_Context
+           (Charset       => +Args.Charset.Get,
+            Unit_Provider => UFP);
```

Comment:
This creates the context before the unit provider is created: IIUC as a result the context will never use the UFP we created below. Can you confirm? If so, it looks like just moving the initalization of Ctx right before iterating of Files should do the job.

Reply:
Hmmm, looks fishy indeed, and since my test doesn't use name resolution *and* all sources are in CWD, it probably wasn't caught.

Wondering if I should add a test for that, but not sure what it should look like ...

Reply:
More specifically the Ada test driver is bad at testing this kind of thing, and I don't really want to add a driver just for that case. Since I plan on using this generic in `nameres`, maybe we can consider the provider will be tested as part of that?

Reply:
Good idea, that would work for me.